libafl/src/stages/calibrate.rs

```diff
@@ -122,7 +122,7 @@ where
 
         let mut start = current_time();
 
-        let exit_kind = executor.run_target(fuzzer, state, mgr, &input)?;
+        let exit_kind = executor.run_target(fuzzer, state, mgr, &mut input.clone())?;
```

Comment:
now in these stages we need to call input.clone() instead because it's mutable

there's a line above this,
```
let input = state.corpus().cloned_input_for_id(corpus_idx)?;
```
which already does the clone, but this clone is unnecessary.
can you change all these lines to?
```
let input = state.corpus().get(idx).borrow().load_input()
```
then we have 1 less clone

Reply:
Yep good point 🫡

Reply:
I don't know if we can do this without two clones, since `state.corpus().get(idx)?.borrow().load_input(state.corpus())` returns `&S::Input` that has an immutable borrow on the state, we can't pass in the `&S::Input` *and* the `&mut State` to `pre_exec_all`...

Reply:
borrow_mut is okay

Reply:
Never mind actually it is ok :) I just mistyped.

Reply:
Yeah, loads of:

```
cannot borrow `*state` as mutable because it is also borrowed as immutable
```

If we try and do that ;_;

Reply:
Also can't clone first and then try to:

```rust
executor.observers_mut().pre_exec_all(
    state,
    state
        .corpus()
        .get(corpus_idx)?
        .borrow()
        .load_input(state.corpus())?,
)?;
```

Reply:
basically
https://github.com/AFLplusplus/LibAFL/pull/1201
cloned_input_for_id() is introduced here.
now that we need clone anyways so I want to stop using this cloned_input... function

Reply:
I'll look if we can do this tomorrow

Reply:
Sounds good :) let me know if I can do anything to help!

Reply:
it seems impossible to get rid of the first clone. i think i just misunderstood it before.
looks good with the current state